### PR TITLE
Init

### PR DESCRIPTION
**Pull request by Scriptkiddi**

### CODE REVIEW

**Comment by dasJ** [comment]
Repo Description & Tags??

File: README.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
+# mattermost-github-integration
+
+This role sets up a mattermost github integration service which posts changes of an organization or repository to a mattermost channel.
```
Comment (dasJ) [inline README.md]: *Mattermost *GitHub

File: README.md [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+
+## Requirements
+
+This role works standalone
```
Comment (dasJ) [inline README.md]: So .... Remove the section?

File: README.md [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+
+## Role Variables
+
+| Name                                           |         Required         | Default                              | Description                                                         |
+|:-----------------------------------------------|:------------------------:|:-------------------------------------|:--------------------------------------------------------------------|
+| `global_cache_dir`                             |    :heavy_check_mark:    |                                      | Cache directory to download roundcube files to                      |
+| `mattermost_github_integration_install_path`   | :heavy_multiplication_x: | `/opt/mattermost-github-integration` | Path to install the server on                                       |
+| `mattermost_github_integration_secret`         | :heavy_multiplication_x: | `None`                               | Secret that is used to authenticate between github and your server. |
```
Comment (SlothOfAnarchy) [inline README.md]: Sure this is not mandatory?

File: README.md [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+
+## Requirements
+
+This role works standalone
+
+
+## Role Variables
+
+| Name                                           |         Required         | Default                              | Description                                                         |
```
Comment (dasJ) [inline README.md]: Also, why don't you merge Required and Default?

File: README.md [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+|:-----------------------------------------------|:------------------------:|:-------------------------------------|:--------------------------------------------------------------------|
+| `global_cache_dir`                             |    :heavy_check_mark:    |                                      | Cache directory to download roundcube files to                      |
+| `mattermost_github_integration_install_path`   | :heavy_multiplication_x: | `/opt/mattermost-github-integration` | Path to install the server on                                       |
+| `mattermost_github_integration_secret`         | :heavy_multiplication_x: | `None`                               | Secret that is used to authenticate between github and your server. |
+| `mattermost_github_integration_server_hook`    | :heavy_multiplication_x: | `/`                                  | The relativ url where the server is listing.                        |
```
Comment (SlothOfAnarchy) [inline README.md]: *relative *listening

File: README.md [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+| `mattermost_github_integration_username`       | :heavy_multiplication_x: | `Github`                             | Name under which username the post should show up in mattermost     |
+| `mattermost_github_integration_icon_url`       | :heavy_multiplication_x: | ` `                                  | Url to icon file which should show up in mattermost                 |
+| `mattermost_github_integration_webhook_urls`   | :heavy_multiplication_x: | `[]`                                 | List of webhooks to post to. See below for more information         |
+| `mattermost_github_integration_show_avatars`   | :heavy_multiplication_x: | `true`                               | show github avatars in the message that is posted                   |
+| `mattermost_github_integration_ignore_actions` | :heavy_multiplication_x: | `[]`                                 | List of actions to ignore- See below for more information           |
+
+### Webhooks
+
+| Name      |      Required      | Default | Description                   |
```
Comment (dasJ) [inline README.md]: This table isn't indented

File: README.md [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+|:-----------------------------------------------|:------------------------:|:-------------------------------------|:--------------------------------------------------------------------|
+| `global_cache_dir`                             |    :heavy_check_mark:    |                                      | Cache directory to download roundcube files to                      |
+| `mattermost_github_integration_install_path`   | :heavy_multiplication_x: | `/opt/mattermost-github-integration` | Path to install the server on                                       |
+| `mattermost_github_integration_secret`         | :heavy_multiplication_x: | `None`                               | Secret that is used to authenticate between github and your server. |
+| `mattermost_github_integration_server_hook`    | :heavy_multiplication_x: | `/`                                  | The relativ url where the server is listing.                        |
+| `mattermost_github_integration_server_address` | :heavy_multiplication_x: | `0.0.0.0`                            | Adress of the server                                                |
```
Comment (SlothOfAnarchy) [inline README.md]: *Address

File: README.md (hunk 3 above)
Comment (dasJ) [inline README.md]: *GitHub

File: README.md (hunk 5 above)
Comment (dasJ) [inline README.md]: *relative URL

File: README.md [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+| `mattermost_github_integration_server_port`    | :heavy_multiplication_x: | `5000`                               | Port under which the server listens to webooks from github          |
```
Comment (SlothOfAnarchy) [inline README.md]: *webhooks

File: README.md (hunk 5 above)
Comment (dasJ) [reply README.md]: Do you mean "listens on"?

File: README.md (hunk 7 above)
Comment (dasJ) [inline README.md]: Not the listening address? Kinda unclear?

File: README.md (hunk 8 above)
Comment (dasJ) [inline README.md]: *GitHub

File: README.md [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+| `mattermost_github_integration_username`       | :heavy_multiplication_x: | `Github`                             | Name under which username the post should show up in mattermost     |
```
Comment (dasJ) [inline README.md]: Name under which username .... Confusing

File: README.md [hunk 10]
```diff
@@ -0,0 +1,74 @@
...
+| `mattermost_github_integration_icon_url`       | :heavy_multiplication_x: | ` `                                  | Url to icon file which should show up in mattermost                 |
```
Comment (dasJ) [inline README.md]: *URL

File: README.md [hunk 11]
```diff
@@ -0,0 +1,74 @@
...
+
+### Webhooks
+
+| Name      |      Required      | Default | Description                   |
+|:----------|:------------------:|:--------|:------------------------------|
+| `name`    | :heavy_check_mark: |         | Name of the repository/team where this information should go to |
+| `url`     | :heavy_check_mark: |         | Url of the mattermost webhook |
+| `channel` | :heavy_check_mark: |         | Channel id like town-square from mattermost. This is where the information gets posted to. |
+
+When using `name` you can add `default` as a variable where everything is send to if it finds no other match. Also you can use `teamname` or `teamname/repositoryname` to have a better controll over the information like `stuvusIT` or `stuvusIT/mattermost-github-integration`
```
Comment (SlothOfAnarchy) [inline README.md]: *control

File: README.md [hunk 12]
```diff
@@ -0,0 +1,74 @@
...
+| `mattermost_github_integration_webhook_urls`   | :heavy_multiplication_x: | `[]`                                 | List of webhooks to post to. See below for more information         |
+| `mattermost_github_integration_show_avatars`   | :heavy_multiplication_x: | `true`                               | show github avatars in the message that is posted                   |
+| `mattermost_github_integration_ignore_actions` | :heavy_multiplication_x: | `[]`                                 | List of actions to ignore- See below for more information           |
+
+### Webhooks
+
+| Name      |      Required      | Default | Description                   |
+|:----------|:------------------:|:--------|:------------------------------|
+| `name`    | :heavy_check_mark: |         | Name of the repository/team where this information should go to |
+| `url`     | :heavy_check_mark: |         | Url of the mattermost webhook |
```
Comment (dasJ) [inline README.md]: *URL *Mattermost

File: README.md [hunk 13]
```diff
@@ -0,0 +1,74 @@
...
+| `channel` | :heavy_check_mark: |         | Channel id like town-square from mattermost. This is where the information gets posted to. |
```
Comment (dasJ) [inline README.md]: *ID

File: README.md (hunk 11 above)
Comment (dasJ) [inline README.md]: *control

File: README.md [hunk 14]
```diff
@@ -0,0 +1,74 @@
...
+
+When using `name` you can add `default` as a variable where everything is send to if it finds no other match. Also you can use `teamname` or `teamname/repositoryname` to have a better controll over the information like `stuvusIT` or `stuvusIT/mattermost-github-integration`
+
+### Ignore actions
+| Name       |      Required      | Default | Description                                             |
+|:-----------|:------------------:|:--------|:--------------------------------------------------------|
+| `category` | :heavy_check_mark: |         | Path to install the server on                           |
+| `actions`  | :heavy_check_mark: |         | List of actions to be ignored e.g `labeled`, `assigned` |
```
Comment (dasJ) [inline README.md]: Is there a complete list somewhere? Also, what kind of description is this for a category?

File: README.md [hunk 15]
```diff
@@ -0,0 +1,74 @@
...
+| `name`    | :heavy_check_mark: |         | Name of the repository/team where this information should go to |
+| `url`     | :heavy_check_mark: |         | Url of the mattermost webhook |
+| `channel` | :heavy_check_mark: |         | Channel id like town-square from mattermost. This is where the information gets posted to. |
+
+When using `name` you can add `default` as a variable where everything is send to if it finds no other match. Also you can use `teamname` or `teamname/repositoryname` to have a better controll over the information like `stuvusIT` or `stuvusIT/mattermost-github-integration`
+
+### Ignore actions
+| Name       |      Required      | Default | Description                                             |
+|:-----------|:------------------:|:--------|:--------------------------------------------------------|
+| `category` | :heavy_check_mark: |         | Path to install the server on                           |
```
Comment (SlothOfAnarchy) [inline README.md]: fix description

File: handlers/main.yml [hunk 16]
```diff
@@ -0,0 +1,7 @@
+---
+- name: Restart mattermost-github-integration service
+  service:
+    name: mattermost-github-integration
+    enabled: yes
+    state: restarted 
+  listen: "restart service"
```
Comment (dasJ) [inline handlers/main.yml]: Why do you even have this listen? Just do `notify: Restart mattermost-github-integration service` and remove the listen

File: meta/main.yml [hunk 17]
```diff
@@ -0,0 +1,23 @@
+---
+galaxy_info:
+  author: Fritz Otlinghaus
+  description: A Bot to post github changes to mattermost
```
Comment (dasJ) [inline meta/main.yml]: *GitHub *Mattermost

File: tasks/main.yml [hunk 18]
```diff
@@ -0,0 +1,31 @@
...
+
+- name: Create mattermost-github-integration systemd service 
+  template:
+    src: systemd_file.j2
+    dest: /etc/systemd/system/mattermost-github-integration.service
+    owner: root
+    group: root
+    mode: 0664
+
+
```
Comment (SlothOfAnarchy) [inline tasks/main.yml]: Remove empty lines

File: meta/main.yml [hunk 19]
```diff
@@ -0,0 +1,23 @@
...
+    - name: Ubuntu
+      versions:
+        - xenial
+        - yakkety
+        - zesty
+        - artful
+
+  galaxy_tags:
+    - mattermost
+    - github
```
Comment (dasJ) [inline meta/main.yml]: Push Notification? Push? Webhook?

File: tasks/main.yml [hunk 20]
```diff
@@ -0,0 +1,31 @@
...
+- name: Install required packages
+  apt:
+    name: "{{ item }}"
+  with_items:
+    - python
+    - python-flask
+    - python-requests
+    - python-pillow
+
+- name: Clone reposity
```
Comment (dasJ) [inline tasks/main.yml]: *repository

File: README.md (hunk 2 above)
Comment (dasJ) [reply README.md]: Btw, Ubuntu is required

File: templates/config.py.j2 [hunk 21]
```diff
@@ -0,0 +1,22 @@
+USERNAME = "{{ mattermost_github_integration_username }}"
+ICON_URL = "{{ mattermost_github_integration_icon_url }}"
+MATTERMOST_WEBHOOK_URLS = {
+    {% for item in mattermost_github_integration_webhook_urls %}
+        '{{ item.name }}' : ("{{ item.url }}", "{{ item.channel }}"),
+    {% endfor %}
+
```
Comment (SlothOfAnarchy) [inline templates/config.py.j2]: empty line

File: tasks/main.yml [hunk 22]
```diff
@@ -0,0 +1,31 @@
...
+  git:
+    repo: 'https://github.com/softdevteam/mattermost-github-integration.git'
+    dest: "{{ mattermost_github_integration_install_path }}"
+    depth: 1
```
Comment (dasJ) [inline tasks/main.yml]: Is `update: yes` the default? Also, please link to the original repo in the README

File: tasks/main.yml [hunk 23]
```diff
@@ -0,0 +1,31 @@
...
+
+- name: Copy config
```
Comment (dasJ) [inline tasks/main.yml]: *configuration file

File: tasks/main.yml [hunk 24]
```diff
@@ -0,0 +1,31 @@
...
+- name: Clone reposity
+  git:
+    repo: 'https://github.com/softdevteam/mattermost-github-integration.git'
+    dest: "{{ mattermost_github_integration_install_path }}"
+    depth: 1
+
+- name: Copy config 
+  template:
+    src: config.py.j2
+    dest: "{{ mattermost_github_integration_install_path }}/mattermostgithub/config.py"
```
Comment (dasJ) [inline tasks/main.yml]: No owner and permissions?

File: tasks/main.yml [hunk 25]
```diff
@@ -0,0 +1,31 @@
...
+  notify: restart service
+
+- name: Create mattermost-github-integration systemd service 
+  template:
+    src: systemd_file.j2
```
Comment (dasJ) [inline tasks/main.yml]: Those names get weirder and weirder

File: templates/config.py.j2 [hunk 26]
```diff
@@ -0,0 +1,22 @@
+USERNAME = "{{ mattermost_github_integration_username }}"
+ICON_URL = "{{ mattermost_github_integration_icon_url }}"
+MATTERMOST_WEBHOOK_URLS = {
+    {% for item in mattermost_github_integration_webhook_urls %}
+        '{{ item.name }}' : ("{{ item.url }}", "{{ item.channel }}"),
```
Comment (SlothOfAnarchy) [inline templates/config.py.j2]: Is this format valid with the trailing comma?

File: tasks/main.yml [hunk 27]
```diff
@@ -0,0 +1,31 @@
...
+    dest: /etc/systemd/system/mattermost-github-integration.service
+    owner: root
+    group: root
+    mode: 0664
+
```
Comment (dasJ) [inline tasks/main.yml]: Please remove this empty line

File: templates/systemd_file.j2 [hunk 28]
```diff
@@ -0,0 +1,10 @@
+[Unit]
+Description=Mattermost Github integration
+After=multi-user.target
+
+[Service]
+Type=simple
+ExecStart=/usr/bin/python {{ mattermost_github_integration_install_path }}/server.py
```
Comment (dasJ) [inline templates/systemd_file.j2]: Hell yeah, let's run this as root!

File: templates/systemd_file.j2 [hunk 29]
```diff
@@ -0,0 +1,10 @@
+[Unit]
+Description=Mattermost Github integration
```
Comment (dasJ) [inline templates/systemd_file.j2]: *GitHub

File: templates/systemd_file.j2 [hunk 30]
```diff
@@ -0,0 +1,10 @@
...
+After=multi-user.target
```
Comment (dasJ) [inline templates/systemd_file.j2]: Just no, this doesn't work

File: templates/config.py.j2 [hunk 31]
```diff
@@ -0,0 +1,22 @@
...
+    {% endfor %}
+
+}
+GITHUB_IGNORE_ACTIONS = {
+    {% for item in mattermost_github_integration_ignore_actions %}
+        "item.category": [{% for action in item.actions %}
+            "{{ action}}",
```
Comment (SlothOfAnarchy) [inline templates/config.py.j2]: space

File: README.md (hunk 3 above)
Comment (Scriptkiddi) [reply README.md]: yes

File: README.md (hunk 5 above)
Comment (Scriptkiddi) [reply README.md]: fixed

File: README.md (hunk 7 above)
Comment (Scriptkiddi) [reply README.md]: fixed

File: README.md (hunk 8 above)
Comment (Scriptkiddi) [reply README.md]: fixed

File: README.md (hunk 11 above)
Comment (Scriptkiddi) [reply README.md]: fixed

File: README.md (hunk 15 above)
Comment (Scriptkiddi) [reply README.md]: fixed

File: tasks/main.yml (hunk 18 above)
Comment (Scriptkiddi) [reply tasks/main.yml]: fixed

File: templates/config.py.j2 (hunk 26 above)
Comment (Scriptkiddi) [reply templates/config.py.j2]: yes

File: templates/config.py.j2 (hunk 21 above)
Comment (Scriptkiddi) [reply templates/config.py.j2]: fixed

File: templates/config.py.j2 (hunk 31 above)
Comment (Scriptkiddi) [reply templates/config.py.j2]: fixed

File: README.md (hunk 1 above)
Comment (Scriptkiddi) [reply README.md]: fixed

File: README.md (hunk 2 above)
Comment (Scriptkiddi) [reply README.md]: fixed

File: README.md (hunk 4 above)
Comment (Scriptkiddi) [reply README.md]: because i like it this way

File: README.md (hunk 3 above)
Comment (Scriptkiddi) [reply README.md]: fixed

File: README.md (hunk 5 above)
Comment (Scriptkiddi) [reply README.md]: fixed

File: README.md (hunk 8 above)
Comment (Scriptkiddi) [reply README.md]: fixed

File: README.md (hunk 9 above)
Comment (Scriptkiddi) [reply README.md]: fixed

File: README.md (hunk 10 above)
Comment (Scriptkiddi) [reply README.md]: fixed

File: README.md (hunk 6 above)
Comment (Scriptkiddi) [reply README.md]: fixed

File: README.md (hunk 12 above)
Comment (Scriptkiddi) [reply README.md]: fixed

File: README.md (hunk 13 above)
Comment (Scriptkiddi) [reply README.md]: fixed

File: README.md (hunk 11 above)
Comment (Scriptkiddi) [reply README.md]: fixed

File: README.md (hunk 14 above)
Comment (Scriptkiddi) [reply README.md]: fixed

File: handlers/main.yml (hunk 16 above)
Comment (Scriptkiddi) [reply handlers/main.yml]: fixed

File: meta/main.yml (hunk 19 above)
Comment (Scriptkiddi) [reply meta/main.yml]: fixed

File: tasks/main.yml (hunk 20 above)
Comment (Scriptkiddi) [reply tasks/main.yml]: fixed

File: tasks/main.yml (hunk 23 above)
Comment (Scriptkiddi) [reply tasks/main.yml]: fixed

File: tasks/main.yml (hunk 27 above)
Comment (Scriptkiddi) [reply tasks/main.yml]: fixed

File: tasks/main.yml (hunk 22 above)
Comment (Scriptkiddi) [reply tasks/main.yml]: its default and fixed

File: templates/systemd_file.j2 (hunk 30 above)
Comment (Scriptkiddi) [reply templates/systemd_file.j2]: why? because it does

File: templates/systemd_file.j2 (hunk 29 above)
Comment (Scriptkiddi) [reply templates/systemd_file.j2]: fixed

**Comment by Scriptkiddi** [comment]
@SlothOfAnarchy @dasJ pls rereview

File: README.md [hunk 32]
```diff
@@ -1,11 +1,11 @@
-# mattermost-github-integration
+# Mattermost-Github-integration
 
 This role sets up a mattermost github integration service which posts changes of an organization or repository to a mattermost channel.
```
Comment (SlothOfAnarchy) [inline README.md]: I think @dasJ actually meant the sentence, not the headline

File: meta/main.yml [hunk 33]
```diff
@@ -19,5 +19,8 @@ galaxy_info:
   galaxy_tags:
     - mattermost
     - github
+    - push
+    - webhook
+    - push notification
```
Comment (SlothOfAnarchy) [inline meta/main.yml]: Have you verified that spaces are allowed in tags?

File: README.md (hunk 32 above)
Comment (Scriptkiddi) [reply README.md]: fixed

File: meta/main.yml (hunk 33 above)
Comment (Scriptkiddi) [reply meta/main.yml]: they are not fixed

**Comment by Scriptkiddi** [comment]
@dasJ @SlothOfAnarchy rereview

File: README.md [hunk 34]
```diff
@@ -0,0 +1,78 @@
+# mattermost-github-integration
+
+This role sets up a Mattermost GitHub integration service which posts changes of an organization or repository to a mattermost channel.
```
Comment (schuppentier) [inline README.md]: *Mattermost

File: README.md [hunk 35]
```diff
@@ -0,0 +1,78 @@
...
+For more information see [softdevteams repository](https://github.com/softdevteam/mattermost-github-integration)
```
Comment (schuppentier) [inline README.md]: Dot at the end of sentences

File: README.md [hunk 36]
```diff
@@ -0,0 +1,78 @@
...
+
+## Requirements
+
+This role requires an apt based system like Ubuntu.
+
```
Comment (schuppentier) [inline README.md]: Remove one empty line

File: README.md [hunk 37]
```diff
@@ -0,0 +1,78 @@
...
+
+## Role Variables
+
+| Name                                           |         Required         | Default                              | Description                                                         |
+|:-----------------------------------------------|:------------------------:|:-------------------------------------|:--------------------------------------------------------------------|
+| `global_cache_dir`                             |    :heavy_check_mark:    |                                      | Cache directory to download roundcube files to                      |
```
Comment (schuppentier) [inline README.md]: Is this a roundcube now?

File: README.md [hunk 38]
```diff
@@ -0,0 +1,78 @@
...
+| `mattermost_github_integration_install_path`   | :heavy_multiplication_x: | `/opt/mattermost-github-integration` | Path to install the server on                                       |
```
Comment (schuppentier) [inline README.md]: The server is hopefully already installed

File: README.md [hunk 39]
```diff
@@ -0,0 +1,78 @@
...
+| `mattermost_github_integration_secret`         | :heavy_multiplication_x: | `None`                               | Secret that is used to authenticate between GitHub and your server. |
```
Comment (schuppentier) [inline README.md]: So now you add dots in the last column? Just remove them

File: README.md [hunk 40]
```diff
@@ -0,0 +1,78 @@
...
+| `mattermost_github_integration_server_port`    | :heavy_multiplication_x: | `5000`                               | Port under which the server listens to webhooks from GitHub         |
+| `mattermost_github_integration_username`       | :heavy_multiplication_x: | `Github`                             | Username to post under in Mattermost                                |
+| `mattermost_github_integration_icon_url`       | :heavy_multiplication_x: | ` `                                  | URL to icon file which should show up in Mattermost                 |
+| `mattermost_github_integration_webhook_urls`   | :heavy_multiplication_x: | `[]`                                 | List of webhooks to post to. See below for more information         |
+| `mattermost_github_integration_show_avatars`   | :heavy_multiplication_x: | `true`                               | Show GitHub avatars in the message that is posted                   |
+| `mattermost_github_integration_ignore_actions` | :heavy_multiplication_x: | `[]`                                 | List of actions to ignore- See below for more information           |
```
Comment (schuppentier) [inline README.md]: Dont you mean `ignore.` instead of `ignore-`?

File: README.md [hunk 41]
```diff
@@ -0,0 +1,78 @@
...
+| `mattermost_github_integration_user`           | :heavy_multiplication_x: | `www-data`                           | User under which the server should run                              |
```
Comment (schuppentier) [inline README.md]: Do you create the user?

Comment (schuppentier) [reply README.md]: If not, please document

File: README.md [hunk 42]
```diff
@@ -0,0 +1,78 @@
...
+| `mattermost_github_integration_group`          | :heavy_multiplication_x: | `www-data`                           | Group under which the server should run                             |
+
+### Webhooks
+
+| Name      |      Required      | Default | Description                                                                                |
+|:----------|:------------------:|:--------|:-------------------------------------------------------------------------------------------|
+| `name`    | :heavy_check_mark: |         | Name of the repository/team where this information should go to                            |
+| `url`     | :heavy_check_mark: |         | URL of the Mattermost webhook                                                              |
+| `channel` | :heavy_check_mark: |         | Channel ID like town-square from mattermost. This is where the information gets posted to. |
```
Comment (schuppentier) [inline README.md]: *Mattermost

File: README.md [hunk 43]
```diff
@@ -0,0 +1,78 @@
...
+
+When using `name` you can add `default` as a variable where everything is send to if it finds no other match. Also you can use `teamname` or `teamname/repositoryname` to have a better control over the information like `stuvusIT` or `stuvusIT/mattermost-github-integration`
+
+### Ignore actions
+| Name       |      Required      | Default | Description                                             |
```
Comment (schuppentier) [inline README.md]: Empty line before the table

File: README.md [hunk 44]
```diff
@@ -0,0 +1,78 @@
...
+          - assigned
+  roles:
+    - mattermost-github-integration
+```
+
+## License
+
+This work is licensed under a [Creative Commons Attribution-ShareAlike 4.0 International License](https://creativecommons.org/licenses/by-sa/4.0/).
+
+
```
Comment (schuppentier) [inline README.md]: Empty line

File: meta/main.yml (hunk 19 above)
Comment (schuppentier) [reply meta/main.yml]: Thanks

File: tasks/main.yml [hunk 45]
```diff
@@ -0,0 +1,33 @@
...
+    - python
+    - python-flask
+    - python-requests
+    - python-pillow
+
+- name: Clone repository
+  git:
+    repo: 'https://github.com/softdevteam/mattermost-github-integration.git'
+    dest: "{{ mattermost_github_integration_install_path }}"
+    depth: 1
```
Comment (schuppentier) [inline tasks/main.yml]: Why don't you add `update: true`? And what if I want to fix the version? Please supply a revision attribute that defaults to master

File: tasks/main.yml [hunk 46]
```diff
@@ -0,0 +1,33 @@
...
+  become: yes
+  become_user: "{{ mattermost_github_integration_user }}"
+
+- name: Copy configuration file
+  template:
+    src: config.py.j2
+    dest: "{{ mattermost_github_integration_install_path }}/mattermostgithub/config.py"
+    owner: "{{ mattermost_github_integration_user }}"
+    group: "{{ mattermost_github_integration_group }}"
```
Comment (schuppentier) [inline tasks/main.yml]: Why don't you set permissions?

File: README.md (hunk 34 above)
Comment (Scriptkiddi) [reply README.md]: fixed

File: README.md (hunk 35 above)
Comment (Scriptkiddi) [reply README.md]: fixed

File: README.md (hunk 36 above)
Comment (Scriptkiddi) [reply README.md]: fixed

File: README.md (hunk 37 above)
Comment (Scriptkiddi) [reply README.md]: fixed

File: README.md (hunk 39 above)
Comment (Scriptkiddi) [reply README.md]: fixed

File: README.md (hunk 38 above)
Comment (Scriptkiddi) [reply README.md]: its the install path for the mattermost-github-integration server

File: README.md (hunk 40 above)
Comment (Scriptkiddi) [reply README.md]: fixed

File: README.md (hunk 41 above)
Comment (Scriptkiddi) [reply README.md]: fixed

File: README.md (hunk 42 above)
Comment (Scriptkiddi) [reply README.md]: fixed

File: README.md (hunk 43 above)
Comment (Scriptkiddi) [reply README.md]: fixed

File: README.md (hunk 44 above)
Comment (Scriptkiddi) [reply README.md]: fixed

File: tasks/main.yml (hunk 46 above)
Comment (Scriptkiddi) [reply tasks/main.yml]: what do you mean?

File: tasks/main.yml (hunk 45 above)
Comment (Scriptkiddi) [reply tasks/main.yml]: fixed

**Comment by Scriptkiddi** [comment]
@dasJ pls rereview

File: README.md [hunk 47]
```diff
@@ -0,0 +1,77 @@
...
+For more information see [softdevteams repository](https://github.com/softdevteam/mattermost-github-integration).
+
+## Requirements
+This role requires an apt based system like Ubuntu.
+
+## Role Variables
+
+| Name                                           | Required                 | Default                              | Description                                                                                                      |
+|:-----------------------------------------------|:------------------------:|:-------------------------------------|:-----------------------------------------------------------------------------------------------------------------|
+| `global_cache_dir`                             | :heavy_check_mark:       |                                      | Cache directory to download mattermost-github-integratoin files to                                               |
```
Comment (dasJ) [inline README.md]: *integration

File: README.md [hunk 48]
```diff
@@ -0,0 +1,77 @@
...
+|:----------|:------------------:|:--------|:-------------------------------------------------------------------------------------------|
+| `name`    | :heavy_check_mark: |         | Name of the repository/team where this information should go to                            |
+| `url`     | :heavy_check_mark: |         | URL of the Mattermost webhook                                                              |
+| `channel` | :heavy_check_mark: |         | Channel ID like town-square from Mattermost. This is where the information gets posted to. |
+
+When using `name` you can add `default` as a variable where everything is send to if it finds no other match. Also you can use `teamname` or `teamname/repositoryname` to have a better control over the information like `stuvusIT` or `stuvusIT/mattermost-github-integration`
+
+### Ignore actions
+
+| Name       |      Required      | Default | Description                                             |
```
Comment (dasJ) [inline README.md]: Why even have a default column in the first place?

File: README.md [hunk 49]
```diff
@@ -0,0 +1,77 @@
...
+| `mattermost_github_integration_ignore_actions` | :heavy_multiplication_x: | `[]`                                 | List of actions to ignore. See below for more information                                                        |
+| `mattermost_github_integration_user`           | :heavy_multiplication_x: | `www-data`                           | User under which the server should run. The user has to exist                                                    |
+| `mattermost_github_integration_group`          | :heavy_multiplication_x: | `www-data`                           | Group under which the server should run. The group hast to exist                                                 |
+| `mattermost_github_integration_version`        | :heavy_multiplication_x: | `master`                             | What version of the repository to check out. This can be the the literal string HEAD, a branch name, a tag name. |
+
+### Webhooks
+
+| Name      |      Required      | Default | Description                                                                                |
```
Comment (dasJ) [inline README.md]: Why even have a default column in the first place?

File: tasks/main.yml (hunk 45 above)
Comment (dasJ) [reply tasks/main.yml]: Why do you always lie to me? :/

File: tasks/main.yml (hunk 46 above)
Comment (dasJ) [reply tasks/main.yml]: `mode: 0640` for example

File: README.md (hunk 47 above)
Comment (Scriptkiddi) [reply README.md]: erledigt

File: README.md (hunk 49 above)
Comment (Scriptkiddi) [reply README.md]: deleted

File: README.md (hunk 48 above)
Comment (Scriptkiddi) [reply README.md]: deleted

**Pull request merged**